### ISSUE TITLE
[Performance]: GPU utilization is low when running large batches on H100

### ISSUE THREAD

**Issue by sleepwalker2017** [vllm-project/vllm#6560]
### Proposal to improve performance

Hi all, I'm running vicuna 13B on H100 using fp8, and I find when batch size is large, say 64 or 96, the gpu utilization is low, about 60%, this is an important cause for the low performance.

I did some analysis, part of this is caused by the schedule and post process of requests. 

Do you have any plans for improving this?

![image](https://github.com/user-attachments/assets/958fb6f3-baef-417b-ba9b-92f39d4bb0db)

### Your current environment (if you think it is necessary)

```text
The output of `python collect_env.py`
```

**Comment by youkaichao** [comment]
definitely, it is listed in https://github.com/vllm-project/vllm/issues/5805

**Comment by sleepwalker2017** [comment]
> definitely, it is listed in #5805

Do you have any idea about how to optimize it?

In my knowledge, python code can only run on 1 cpu core because of GIL, so how can we parallelize the post process and the decoding process ? 

I'll spend some time trying to improve this, do you have any advice?  thank you.

**Comment by VincentXWD** [comment]
@sleepwalker2017 btw could you please share the command when profiling vLLM using Nsight?
Thanks!

**Comment by sleepwalker2017** [comment]
> @sleepwalker2017 btw could you please share the command when profiling vLLM using Nsight?

The command is nothing special, I think it's only `nsys profile python xxx.py`. you can refer to nsys manual to see the usage.

**Comment by VincentXWD** [comment]
> > @sleepwalker2017 btw could you please share the command when profiling vLLM using Nsight?
> 
> The command is nothing special, I think it's only `nsys profile python xxx.py`. you can refer to nsys manual to see the usage.

Thanks for your quick reply. I see. It seems that you were using nsys to profile a single .py script. I thought that you were benchmarking the service.

**Comment by artetaout** [comment]
For  benchmarking the server, I will add nvtx marker to control the code start and end domain, just before and after .step
